cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -212,6 +330,49 @@ impl AuthorizationCall {
         let (policies, entities) = self.slice.try_into(schema.as_ref())?;
         Ok((q, policies, entities))
     }
+
+    #[cfg(feature = "partial-eval")]
+    fn get_components_partial(self) -> Result<(Request, PolicySet, Entities), Vec<String>> {
+        let schema = self
+            .schema
+            .map(|v| Schema::from_json_value(v.into()))
+            .transpose()
+            .map_err(|e| [e.to_string()])?;
+        let principal = match self.principal {
+            Some(p) => Some(
+                EntityUid::from_json(p.into())
+                    .map_err(|e| ["Failed to parse principal".into(), e.to_string()])?,
+            ),
+            None => None,
+        };
+        let action = EntityUid::from_json(self.action.into())
+            .map_err(|e| ["Failed to parse action".into(), e.to_string()])?;
+        let resource = match self.resource {
+            Some(r) => Some(
+                EntityUid::from_json(r.into())
+                    .map_err(|e| ["Failed to parse resource".into(), e.to_string()])?,
+            ),
+            None => None,
+        };
+
+        let context = serde_json::to_value(self.context)
+            .map_err(|e| [format!("Error encoding the context as JSON: {e}")])?;
+        let context = Context::from_json_value(context, schema.as_ref().map(|s| (s, &action)))
+            .map_err(|e| [e.to_string()])?;
+        let mut b = Request::builder().action(Some(action)).context(context);
+        if principal.is_some() {
+            b = b.principal(principal)
+        }
+        if resource.is_some() {
+            b = b.resource(resource)
+        }
+        if self.enable_request_validation && schema.is_some() {
+            b = b.schema(schema.as_ref().unwrap());
```

Comment:
regarding one of the clippy errors in CI: This can `match` on `schema` to avoid the `unwrap()`.